packages/core/src/util/renderer.ts

```diff
@@ -161,11 +165,32 @@ export const createDefaultValue = (schema: JsonSchema) => {
       return [];
     case 'null':
       return null;
+    case 'object':
+      return extractDefaults(schema);
     default:
       return {};
   }
 };
 
+/**
+ * Returns the default value defined in the given schema.
+ * @param schema the schema for which to create a default value.
+ * @returns {any}
+ */
+export const extractDefaults = (schema: JsonSchema) => {
+  if (schema.type === 'object' && schema.default === undefined) {
```

Comment:
The `type === 'object'` is optional in JSON Schema, so we should not rely on it. We have a util method somewhere determining the type of JSON (sub) schemas.